types/block_test.go

```diff
@@ -229,3 +233,164 @@ func TestBlockJsonMarshal(t *testing.T) {
 	AssertHaveSameCid(t, &child, &unmarshalled)
 	assert.True(t, child.Equals(&unmarshalled))
 }
+
+func TestDataToSign(t *testing.T) {
+	tf.UnitTest(t)
+	newAddress := address.NewForTestGetter()
+	
+	b := &Block{
+		Miner:           newAddress(),
+		Tickets:         []Ticket{{VRFProof: []byte{0x01, 0x02, 0x03}}},
+		Height:          Uint64(2),
+		Messages:        CidFromString(t, "somecid"),
+		MessageReceipts: CidFromString(t, "somecid"),
+		Parents:         NewTipSetKey(CidFromString(t, "somecid")),
+		ParentWeight:    Uint64(1000),
+		ElectionProof:   []byte{0x1},
+		StateRoot:       CidFromString(t, "somecid"),
+		Timestamp:       Uint64(1),
+		BlockSig:        []byte{0x3},
+	}
+
+	diff := &Block{
+		Miner:           newAddress(),
+		Tickets:         []Ticket{{VRFProof: []byte{0x03, 0x01, 0x02}}},
+		Height:          Uint64(3),
+		Messages:        CidFromString(t, "someothercid"),
+		MessageReceipts: CidFromString(t, "someothercid"),
+		Parents:         NewTipSetKey(CidFromString(t, "someothercid")),
+		ParentWeight:    Uint64(1001),
+		ElectionProof:   []byte{0x2},
+		StateRoot:       CidFromString(t, "someothercid"),
+		Timestamp:       Uint64(4),
+		BlockSig:        []byte{0x4},		
+	}
+
+	// Changing BlockSig does not affect output
+	func() {
+		before := b.DataToSign()
+		
+		cpy := b.BlockSig
+		defer func(){ b.BlockSig = cpy}()
+		
+		b.BlockSig = diff.BlockSig
+		after := b.DataToSign()
+		assert.True(t, bytes.Equal(before, after))
+	}()
+
+	// Changing all other fields does affect output
+	// Note: using reflectors doesn't seem to make this much less tedious
+	// because it appears that there is no generic field setting function.
+	func() {
+		before := b.DataToSign()
+		
+		cpy := b.Miner
+		defer func(){ b.Miner = cpy}()
+		
+		b.Miner = diff.Miner
+		after := b.DataToSign()
+		assert.False(t, bytes.Equal(before, after))
+	}()
+
+	func() {
```

Comment:
It felt nice to separate each case into a function call and use defer to reset data each time.  Might be more verbose than necessary 🤷‍♂

Reply:
I think you could also use `t.Run()`